dom0/sd-whonix-hidserv-key.sls

```diff
@@ -4,7 +4,24 @@
 {% import_json "sd/config.json" as d %}
 
 # add hidden service auth key to torrc
+{% if d.hidservv3.hostname == "ADD_YOUR_VALUE" %}
 sd-whonix-hidserv-key:
   file.append:
     - name: /usr/local/etc/torrc.d/50_user.conf
-    - text: HidServAuth {{ d.hidserv.hostname }} {{ d.hidserv.key }}
+    - text: HidServAuth {{ d.hidserv.hostname }}.onion {{ d.hidserv.key }}
```

Comment:
I think it might be best to keep .onion in the hostname itself, as a random string of characters might be confusing without the .onion suffix, and will help maintainability and avoid confusion. Since the hostname without .onion is a special case for the v3 auth_private, you can use jinja conditionals before `install-sd-whonix-tor-private-key` to create and using a separate variable might work:
```
{% set hostname_without_dot_onion = d.hidservv3.hostname.split('.')[0] %}
```